runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -154,14 +154,17 @@ public void updateExecutionStatus(Long stateMachineId,Long taskId, Status status
      */
     public void unsidelineState(Long stateMachineId, Long stateId) {
         State state = this.statesDAO.findById(stateId);
+        StateMachine stateMachine = retrieveStateMachine(stateMachineId);
```

Comment:
Here we can make a single call to DB to fetch state machine and after that we can loop through Set<State> of state machine to obtain state object.